### PR TITLE
[SPARK-7672][CORE] Use int conversion in translating kryoserializer.buffer.mb to kryoserializer.buffer

### PR DESCRIPTION
**Pull request by nishkamravi2**
In translating spark.kryoserializer.buffer.mb to spark.kryoserializer.buffer, use of toDouble will lead to "Fractional values not supported" error even when spark.kryoserializer.buffer.mb is an integer.
@ilganeli, @andrewor14

### CODE REVIEW

**Comment by andrewor14** [comment]
ok to test

**Comment by andrewor14** [comment]
This fix looks fine. Thanks for fixing it. Can you verify @ilganeli?

**Comment by ilganeli** [comment]
@andrewor14 @nishkamravi2 Good catch. The fix looks good. Thanks!

**Comment by nishkamravi2** [comment]
Thanks for verifying @andrewor14 @ilganeli. Looked at the failed tests and they seem unrelated.

**Comment by andrewor14** [comment]
Yes, pretty sure they're not related. retest this please

**Comment by vanzin** [comment]
LGTM, but could you add a test for this? Something like this, in `SparkConfSuite::test("deprecated configs")`:

    conf.set("spark.kryoserializer.buffer.max.mb", "1.1")
    assert(conf.getSizeAsKb("spark.kryoserializer.buffer") === 1100)

**Comment by nishkamravi2** [comment]
Good suggestion @vanzin.

File: core/src/test/scala/org/apache/spark/SparkConfSuite.scala [hunk 1]
```diff
@@ -241,6 +241,9 @@ class SparkConfSuite extends FunSuite with LocalSparkContext with ResetSystemPro
 
     conf.set("spark.yarn.applicationMaster.waitTries", "42")
     assert(conf.getTimeAsSeconds("spark.yarn.am.waitTime") === 420)
+
+    conf.set("spark.kryoserializer.buffer.max.mb", "1.1")
```
Comment (vanzin) [inline core/src/test/scala/org/apache/spark/SparkConfSuite.scala]: I expect the test to fail, but my example code was bad (copy & pasted the wrong property). This should be `spark.kryoserializer.buffer.mb`. Sorry about that.

Comment (nishkamravi2) [inline core/src/test/scala/org/apache/spark/SparkConfSuite.scala]: And I thought I had verified your example code, skipped both of us.

**Pull request closed without merging**